libs/langchain/langchain/document_loaders/recursive_url_loader.py

```diff
@@ -1,11 +1,30 @@
-from typing import Iterator, List, Optional, Set
+import re
+from typing import Callable, Iterator, List, Optional, Set, Union
 from urllib.parse import urljoin, urlparse
 
 import requests
 
 from langchain.docstore.document import Document
 from langchain.document_loaders.base import BaseLoader
 
+try:
```

Comment:
We shouldn't be printing on the outer level here - would want to include when we initialize the class only